### DISCUSSION TITLE
How to fill different colors based on the values of the isometric surface

### DISCUSSION THREAD

**Discussion by lucinelouhi** [Mapsui/Mapsui#2619] [Q&A]
May I ask what is the difference and order between a layer's Style and a Feature's Style when rendering? I want to achieve different colors for filling isosurfaces, how can I do better. In addition, when changing the layer data source, it seems that the original data has not been cleared. You can directly set the layer data source, or remove the layer and add another layer. What I want is to directly change the data source of the layer, but it seems that the original data has not disappeared. thanks
﻿

**Comment by pauldendulk** [comment]
> what is the **difference** between a layer's Style and a Feature's Style when rendering?

Look at my answer here: https://github.com/Mapsui/Mapsui/discussions/2621#discussioncomment-9366291

> what is the **order** between a layer's Style and a Feature's Style when rendering?

First Layers then Feature. Look at the implementation here: https://github.com/Mapsui/Mapsui/blob/main/Mapsui/Rendering/VisibleFeatureIterator.cs

> I want to achieve different colors for filling isosurfaces, how can I do better

What kind of data do you have?

> When changing the layer data source, it seems that the original data has not been cleared.

That should work. What type of layer? 'Layer'?
